### PR TITLE
My VA: remove show_myva_dashboard_2_0 feature toggle

### PR DESCRIPTION
**Pull request by dellerbie**
Removes the `show_myva_dashboard_2_0` feature flag and related code since we are 100% enabled for all users.

## Summary

- Remove `show_myva_dashboard_2_0` feature flag
- Remove conditionals using this flag
- Update specs

## Related issue(s)

- [60360](https://github.com/department-of-veterans-affairs/va.gov-team/issues/60360)

## What areas of the site does it impact?
My VA Dashboard

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: src/applications/personalization/dashboard/components/Dashboard.jsx [hunk 1]
```diff
@@ -233,30 +232,21 @@ const Dashboard = ({
...
-                )}
-              {isLOA3 && shouldShowV2Dashboard ? (
-                <EducationAndTraining />
-              ) : null}
-              {isLOA3 && shouldShowV2Dashboard ? (
-                <SavedApplications />
-              ) : (
-                <ApplyForBenefits />
+              {isLOA3 && (
+                <>
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/personalization/dashboard/components/Dashboard.jsx:240]: They call this a React 'fragment' and it is used to wrap two components or elements.

https://react.dev/reference/react/Fragment

It is needed in this instance so you are good to go.

[the remark this answers is not in the document]
Comment (allisonlu) [reply src/applications/personalization/dashboard/components/Dashboard.jsx:240]: We should keep it, React will throw an error if there isn't a parent element wrapping `DebtsV2` and `BenefitPaymentsV2`

**Review by adamwhitlock1 — APPROVED** [review]
I think this all looks good to me. Yay for removing code!

We may want to think about renaming some of the v2 stuff to remove the 'v2' since it is fully launched now and the default.

**Review by allisonlu — APPROVED** [review]
lgtm!

**Review by pjhill — APPROVED** [review]
Nice -- thanks for removing an unneeded feature flag and updating test specs!

**Pull request merged**